ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1330,6 +1332,9 @@ void close() throws StoreException {
     long startTimeInMs = time.milliseconds();
     try {
       persistor.write();
+      if (persistorTask != null) {
```

Comment:
What does it mean if `persistorTask` is null? It can only happen if `scheduler` passed to the constructor is null. However, if that is null, it seems that we construct the `PersistentIndex` just fine without creating a background thread to persist. How is that okay?

Reply:
Never mind - there are use cases where the persisting is done via explicit `persistIndex()` call (like for the target index for compaction) or at shutdown, which do not require a background thread.